src/main/java/com/epam/DataCSVFileWriter.java

```diff
@@ -0,0 +1,77 @@
+package com.epam;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+
+import org.apache.commons.csv.CSVPrinter;
+
+import com.epam.data.RoadAccident;
+import com.epam.data.RoadAccident.DAYTIMETYPE;
+
+public class DataCSVFileWriter implements Runnable {
+	private BlockingQueue<List<RoadAccident>> dataQueue;
+	private CSVPrinter csvPrinterDayTime;
+	private CSVPrinter csvPrinterNightTime;
+
+	public DataCSVFileWriter(BlockingQueue<List<RoadAccident>> dataQueue,
+			CSVPrinter csvPrinterDayTime, CSVPrinter csvPrinterNightTime) {
+		this.dataQueue = dataQueue;
+		this.csvPrinterDayTime = csvPrinterDayTime;
+		this.csvPrinterNightTime = csvPrinterNightTime;
+	}
+
+	// TODO impl the logic to write data into outputstream or use csv writer
+	@Override
+	public void run() {
+
+		List<RoadAccident> consumedData = null;
+		try {
+			consumedData = dataQueue.take();
+
+			while (consumedData != null && !consumedData.isEmpty()) {				
+				consumedData = dataQueue.take();
```

Comment:
I'm pretty sure this line should be at the very end of "while" loop - you are cureently ignoring first batch in your queue.